tidy3d/plugins/autograd/invdes/filters.py

```diff
@@ -5,28 +5,69 @@
 
 from ..functions import convolve
 from ..types import KernelType, PaddingType
-from ..utilities import make_kernel
+from ..utilities import get_kernel_size_px, make_kernel
+
+
+def _get_kernel_size(
+    radius: Union[float, Tuple[float, ...]],
+    dl: Union[float, Tuple[float, ...]],
+    size_px: Union[int, Tuple[int, ...]],
+) -> Tuple[int, ...]:
+    """Determine the kernel size based on the provided radius, grid spacing, or size in pixels.
+
+    Parameters
+    ----------
+    radius : Union[float, Tuple[float, ...]]
+        The radius of the kernel. Can be a scalar or a tuple.
+    dl : Union[float, Tuple[float, ...]]
+        The grid spacing. Can be a scalar or a tuple.
+    size_px : Union[int, Tuple[int, ...]]
+        The size of the kernel in pixels for each dimension. Can be a scalar or a tuple.
+
+    Returns
+    -------
+    Tuple[int, ...]
+        The size of the kernel in pixels for each dimension.
+
+    Raises
+    ------
+    ValueError
+        If neither 'size_px' nor both 'radius' and 'dl' are provided.
+    """
+    if size_px is not None:
+        return (size_px,) if np.isscalar(size_px) else tuple(size_px)
+    elif radius is not None and dl is not None:
+        kernel_size = get_kernel_size_px(radius=radius, dl=dl)
+        return (kernel_size,) if np.isscalar(kernel_size) else tuple(kernel_size)
+    else:
+        raise ValueError("Either 'size_px' or both 'radius' and 'dl' must be provided.")
 
 
 def make_filter(
-    filter_type: KernelType,
-    size: Union[int, Tuple[int, ...]],
+    radius: Union[float, Tuple[float, ...]] = None,
+    dl: Union[float, Tuple[float, ...]] = None,
     *,
+    size_px: Union[int, Tuple[int, ...]] = None,
     normalize: bool = True,
     padding: PaddingType = "reflect",
+    filter_type: KernelType,
 ):
     """Create a filter function based on the specified kernel type and size.
 
     Parameters
     ----------
-    filter_type : KernelType
-        The type of kernel to create (`circular` or `conic`).
-    size : Union[int, Tuple[int, ...]]
-        The size of the kernel in pixels for each dimension. Can be a scalar or a tuple.
+    radius : Union[float, Tuple[float, ...]], optional
```

Comment:
guess we can test whether this will render well in our docs, it might be inconsistent throughout but we generally have been doing `= None`

Reply:
Ah damn yeah, can we keep it in this PR and if it's a problem I'll change it everywhere in another PR?

Reply:
Ok, this is done for everything within the autograd plugin at least

Reply:
cool thanks